src/anon-control-client.ts

```diff
@@ -0,0 +1,350 @@
+import * as net from 'net';
+
+export class AnonControlClient {
+    private client: net.Socket;
+
+    constructor(host = '127.0.0.1', port = 9051) {
+        console.log('Connecting to Anon Control Port at', host, port);
+        
+        this.client = net.createConnection({ host, port }, () => {
+            console.log('Successfully connected to Anon Control Port');
+        });
+    }
+    
+    /**
+     * AUTHENTICATE
+        Sent from the client to the server.  The syntax is:
+
+            "AUTHENTICATE" [ SP 1*HEXDIG / QuotedString ] CRLF
+
+        This command is used to authenticate to the server. The provided string is
+        one of the following:
+
+            * (For the HASHEDPASSWORD authentication method; see 3.21)
+            The original password represented as a QuotedString.
+
+            * (For the COOKIE is authentication method; see 3.21)
+            The contents of the cookie file, formatted in hexadecimal
+
+            * (For the SAFECOOKIE authentication method; see 3.21)
+            The HMAC based on the AUTHCHALLENGE message, in hexadecimal.
+
+        The server responds with 250 OK on success or 515 Bad authentication if
+        the authentication cookie is incorrect.  Tor closes the connection on an
+        authentication failure.
+
+        The authentication token can be specified as either a quoted ASCII string,
+        or as an unquoted hexadecimal encoding of that same string (to avoid escaping
+        issues).
+
+        For information on how the implementation securely stores authentication
+        information on disk, see section 5.1.
+
+        Before the client has authenticated, no command other than
+        PROTOCOLINFO, AUTHCHALLENGE, AUTHENTICATE, or QUIT is valid.  If the
+        controller sends any other command, or sends a malformed command, or
+        sends an unsuccessful AUTHENTICATE command, or sends PROTOCOLINFO or
+        AUTHCHALLENGE more than once, Tor sends an error reply and closes
+        the connection.
+
+        To prevent some cross-protocol attacks, the AUTHENTICATE command is still
+        required even if all authentication methods in Tor are disabled.  In this
+        case, the controller should just send "AUTHENTICATE" CRLF.
+        (Versions of Tor before 0.1.2.16 and 0.2.0.4-alpha did not close the
+        connection after an authentication failure.)
+     * @param password 
+     * @returns 
+     */
+    authenticate(password: string = 'password'): Promise<void> {
```

Comment:
This default is redundant as of now as the user cannot set a password themselves in the anonrc (through the code)

Issue has been discussed and will be resolved in another task